messaging/src/main/java/org/axonframework/configuration/DecoratedComponent.java

```diff
@@ -70,11 +70,25 @@ public D doResolve(@Nonnull Configuration configuration) {
             return existingInstance;
         }
         synchronized (this) {
-            return instanceReference.updateAndGet(instance -> instance != null ? instance :
-                    decorator.decorate(configuration,
-                                       identifier().name(),
-                                       delegate.resolve(configuration))
-
+            return instanceReference.updateAndGet(
```

Comment:
Did we make the mistake in the Spring Context impl :O ? 

```
HierarchicalSpringContextTest.setUp:60 » BeanCreation Error creating bean with name 'commandBus' defined in org.axonframework.springboot.autoconfig.HierarchicalSpringContextTest$ParentContext: Original component type [class org.axonframework.commandhandling.SimpleCommandBus] is not assignable to decorated component type [class org.axonframework.commandhandling.distributed.DistributedCommandBus]. Make sure decorators return matching components, as component retrieval otherwise fails!
```

Reply:
Took some digging, but I've fixed it now. Wondering what Spring-guru @abuijze thanks of the changes in the `SpringComponentRegistry` though...perhaps this is against regular Spring practices.